src/gc.c

```diff
@@ -55,10 +55,6 @@ pagetable_t memory_map;
 bigval_t *big_objects_marked = NULL;
 
 // finalization
-// `ptls->finalizers` and `finalizer_list_marked` might have tagged pointers.
```

Comment:
Why does the optimization has to be removed?

Reply:
I guess it doesn't have to be.

Reply:
The place that you need to handle tagged pointer will be [here](https://github.com/JuliaLang/julia/blob/521dec9bf4990a0ec0fb827e1167e6c5195d392d/src/gc.c#L1869).

Reply:
Thanks.